gui/src/hooks/firmware-update.ts

```diff
@@ -87,18 +78,15 @@ export async function fetchCurrentFirmwareRelease(): Promise<FirmwareRelease | n
   const processedReleses = [];
   for (const release of releases) {
     const fwAsset = firstAsset(release.assets, 'BOARD_SLIMEVR-firmware.bin');
-    const deployAsset = firstAsset(release.assets, 'deploy.json');
     if (!release.assets || !fwAsset /* || release.prerelease */) continue;
 
     let version = release.tag_name;
     if (version.charAt(0) === 'v') {
       version = version.substring(1);
     }
 
-    const userCanUpdate = !deployAsset
-      ? true
-      : await checkUserCanUpdate(deployAsset?.browser_download_url);
-
+    const deployAsset = firstAsset(release.assets, 'deploy.json');
+    const userCanUpdate = await checkUserCanUpdate(deployAsset?.browser_download_url);
```

Comment:
i have an eslint rule on the eslint PR failing from having an await inside this for loop, can you use `Promise.all()`?